### PR TITLE
Updating NativeHost.PlatformToolset version

### PR DESCRIPTION
**Pull request by SergeySmirnov-Akvelon**
## Proposed changes
- Updated NativeHost.PlatformToolset version

## Customer Impact
-  No

## Regression? 
- No

## Risk
- Minimal

## Test environment(s) 
- Microsoft Windows [Version 10.0.19041.388]
- .NET Core SDK 6.0.100-rc.1.21430.12

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/5797)

### CODE REVIEW

File: src/System.Windows.Forms/tests/IntegrationTests/NativeHost/NativeHost.vcxproj [hunk 1]
```diff
@@ -28,29 +28,29 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'" Label="Configuration">
     <ConfigurationType>Application</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
-    <PlatformToolset>v142</PlatformToolset>
+    <PlatformToolset>v143</PlatformToolset>
```
Comment (dreddy-work) [inline src/System.Windows.Forms/tests/IntegrationTests/NativeHost/NativeHost.vcxproj:31]: Isn't this restrict VS to VS 2022? can we still use older VS versions here?

Comment (RussKie) [reply src/System.Windows.Forms/tests/IntegrationTests/NativeHost/NativeHost.vcxproj:31]: .NET 6.0 is only supported in VS 2022

**Review by RussKie — CHANGES_REQUESTED** [review]
Docs must also be updated to reflect all necessary workloads and tools.
E.g. https://github.com/dotnet/winforms/blob/main/docs/developer-guide.md, but mya be other ones as well.

**Comment by SergeySmirnov-Akvelon** [comment]
@RussKie. Added necessary changes to the documentation

**Review by RussKie — APPROVED** [review]

File: src/System.Windows.Forms/tests/IntegrationTests/NativeHost/NativeHost.vcxproj (hunk 1 above)
Comment (dreddy-work) [reply src/System.Windows.Forms/tests/IntegrationTests/NativeHost/NativeHost.vcxproj:31]: Are we saying community can't build our repo without VS 2022? @SergeySmirnov-Akvelon, Did you try building repo with this change but no VS 2022?

Comment (RussKie) [reply src/System.Windows.Forms/tests/IntegrationTests/NativeHost/NativeHost.vcxproj:31]: That is correct, and it has been the case for some months.

Comment (dreddy-work) [reply src/System.Windows.Forms/tests/IntegrationTests/NativeHost/NativeHost.vcxproj:31]: Will take this discussion offline.

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**